index.bs

```diff
@@ -12995,207 +12872,301 @@ with the following characteristics:
+    from a [=/map=] |map|
+    and a |kind| which is either "<code>key+value</code>", "<code>key</code>", or "<code>value</code>"):
+
+    1. Let |closure| be a new [=/Abstract Closure=] with no parameters
+        that captures |map| and |kind| and performs the following steps when called:
+        1. [=map/For each=] |key| → |value| of |map|:
+            1. Set |key| and |value| to each [=converted to an ECMAScript value=].
+            1. If |kind| is "<code>key</code>", let |result| be |key|.
+            1. Else if |kind| is "<code>value</code>", let |result| be |value|.
+            1. Else, let |result| be [$CreateArrayFromList$](« |key|, |value| »).
+            1. Perform [=?=] [$GeneratorYield$]([$CreateIterResultObject$](|result|, <emu-value>false</emu-value>)).
+
+            Note: The [=map/size=] of |map|, and the order of its entries,
+            might have changed while execution of this abstract operation
+            was paused by [$Yield$].
+        1. Return <emu-value>undefined</emu>.
+    1. Return [$CreateIteratorFromClosure$](|closure|, "%MapIteratorPrototype%", {{%MapIteratorPrototype%}}).
+</div>
 
-<h5 id="es-map-keys-values">keys and values</h5>
 
-For both of <code class="idl">keys</code> and <code class="idl">values</code>, there must exist a data property with that name on
+<h5 id="es-map-entries">entries</h5>
+
+There must exist an <code class="idl">entries</code> data property on
 |A|’s [=interface prototype object=]
 with the following characteristics:
 
 *   The property has attributes { \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }.
-*   The value of the property is a [=built-in function object=] that [=forwards to the internal map object|forwards that name to the internal map object=].
+*   The value of the property is a [=built-in function object=]
+    whose behavior when invoked is as follows:
+
+    <div algorithm="to invoke the entries method of Maps">
+        1. Let |O| be the <emu-val>this</emu-val> value, [=implementation-checked=] against <var ignore>a</var> with identifier "<code>entries</code>" and type "<code>method</code>".
+        1.  Let |map| be the [=map entries=] of the IDL value
+            that represents a reference to |O|.
+        1. Return the result of [=creating a map iterator=] from |map| with kind "<code>key+value</code>".
+    </div>
 
-The value of the [=function objects=]’ <code class="idl">length</code> properties is the Number value <emu-val>0</emu-val>.
+The value of the [=function object=]'s <code class="idl">length</code> properties is the Number value <emu-val>0</emu-val>.
 
-The value of the [=function object=]’s <code class="idl">name</code> property is the String value "<code>keys</code>" or "<code>values</code>", correspondingly.
+The value of the [=function object=]’s <code class="idl">name</code> property is the String value "<code>entries</code>".
```

Comment:
It isn't really your mess to clean up but now we have inconsistent smart vs. not-smart quotes on these clauses... if you wanted to make them consistent (probably not-smart?) then that would make things slightly nicer.

Reply:
Eh just took a few minutes to review all the existing smart quotes and replace them with apostrophes. They were indeed wildly inconsistent.